src/components/emailerFilterControls/index.tsx

```diff
@@ -0,0 +1,235 @@
+}
+
+function formatDates(
+  start?: string,
+  end?: string,
+): [moment.Moment | null, moment.Moment | null] | undefined {
+  if (start && end) return [moment(start), moment(end)];
+}
+
+function disabledDate(current: moment.Moment): boolean {
+  // Can not select future days
+  return current > moment().endOf('day');
+}
+
+const activityCountLabels: SliderMarks = {
+  0: '0',
+  [MAX_COUNT + 1]: `${MAX_COUNT}+`,
+};
+
+const neighborhoodOptions = Object.values(Neighborhoods)
+  .map((value) => {
+    return { value };
+  })
+  .sort();
+
+const EmailerFilterControls: React.FC<EmailerFilterControlsProps> = ({
+  filters,
+  setFilters,
+}) => {
+  const [neighborhoodSearch, setNeighborhoodSearch] = useState<string>('');
+  const [namesSearch, setNamesSearch] = useState<string>('');
+
+  const [commonNameOptions, setCommonNameOptions] = useState<
+    { value: string }[]
+  >([]);
+
+  useEffect(() => {
+    async function getNames() {
+      const names = await apiClient.getAllCommonNames();
+      setCommonNameOptions(
+        names.map((name) => {
+          return { value: name };
+        }),
+      );
+    }
+    getNames();
+  }, []);
+
+  const onAddFilter = useCallback(
+    (
+      key: 'neighborhoods' | 'commonNames',
+      options: { value: string }[],
+      searchValue: string,
+      warning: string,
+    ) => {
+      return () => {
+        if (options.map((option) => option.value).includes(searchValue)) {
+          if (!filters[key].includes(searchValue)) {
+            setFilters({ ...filters, [key]: [...filters[key], searchValue] });
+            key === 'neighborhoods'
+              ? setNeighborhoodSearch('')
+              : setNamesSearch('');
+          }
+        } else {
+          message.warn(warning);
+        }
+      };
+    },
+    [filters, setFilters],
+  );
+
+  return (
+    <StyledCollapse ghost={true}>
+      <Collapse.Panel header="Activity Count" key="activityCount">
+        <p>{`${filters.activityCountMin} - ${
+          filters.activityCountMax || MAX_COUNT + '+'
+        }`}</p>
+        <Slider
+          range={true}
+          marks={activityCountLabels}
+          tooltip={{ open: false }}
+          min={0}
+          max={MAX_COUNT + 1}
+          value={activityCountRange(filters)}
+          onChange={([min, max]) => {
+            setFilters({
+              ...filters,
+              activityCountMin: min,
+              activityCountMax: max > MAX_COUNT ? undefined : max,
+            });
+          }}
+        />
+      </Collapse.Panel>
+      <Collapse.Panel header="Adoption Date" key="adoptionDate">
+        <StyledRangePicker
+          value={formatDates(filters.adoptedStart, filters.adoptedEnd)}
+          onChange={(_, dateStrings) =>
+            setFilters({
+              ...filters,
+              adoptedStart: dateStrings[0],
+              adoptedEnd: dateStrings[1],
+            })
+          }
+          disabledDate={disabledDate}
+        />
+        <Button
```

Comment:
❓Does the built-in x button in the range picker not clear the start/end dates correctly? If so, we might want to hide this (if it isn't too difficult)

![image](https://user-images.githubusercontent.com/33704204/232180969-c4851c53-bff8-4c08-a81f-4ac78ffc6a28.png)

Reply:
I didn't even notice that - looks like it's just an [additional boolean flag to hide it](https://ant.design/components/date-picker#common-api)